packages/util/src/bytes.ts

```diff
@@ -25,19 +25,19 @@ export const bytesToUnprefixedHex = _bytesToUnprefixedHex
  * @throws If the input is not a valid 0x-prefixed hex string
  */
 export const hexToBytes = (hex: string) => {
-  if (!hex.startsWith('0x')) throw EthereumJSErrorWithoutCode('input string must be 0x prefixed')
+  if (!isHexString(hex)) throw EthereumJSErrorWithoutCode('input string must be 0x prefixed')
   return nobleH2B(padToEven(stripHexPrefix(hex)))
 }
 
 export const unprefixedHexToBytes = (hex: string) => {
-  if (hex.startsWith('0x')) throw EthereumJSErrorWithoutCode('input string cannot be 0x prefixed')
+  if (isHexString(hex)) throw EthereumJSErrorWithoutCode('input string cannot be 0x prefixed')
   return nobleH2B(padToEven(hex))
 }
 
 export const bytesToHex = (bytes: Uint8Array): PrefixedHexString => {
   if (bytes === undefined || bytes.length === 0) return '0x'
```

Comment:
Unrelated but why do we return `0x` if the input is `undefined`!? :exploding_head:

Reply:
This is weird. I will look into it, I have a separate PR that addresses some types in this file

Reply:
This is actually an error and shouldn't be there in the first place, the input type does not even allow undefined. 

I also looked at the internals of noble hash and it already returns `0x` for empty Uint8Arrays, so there is no need for us to do that manually. https://github.com/ethereumjs/ethereumjs-monorepo/pull/3995 will take care of this